colossus/src/main/scala/colossus/core/IOSystem.scala

```diff
@@ -77,6 +115,14 @@ case class IOSystem private[colossus](workerManager: ActorRef, config: IOSystemC
    */
   val namespace = MetricAddress.Root / config.name
 
+  //ENSURE THIS IS THE LAST THING INITIALIZED!!!
+  private[colossus] val workerManager : ActorRef = managerFactory(workers, this)
+
+  // >[*]< SUPER HACK ALERT >[*]<
+  while (config.numWorkers > 0 && workers().size == 0) {
```

Comment:
Hrm...yea....the only other way I could see of doing this, is to have this class be construcable only via Future.  You would need some intermediate object which was checking the state of the Workers, and then filling a Promise when they are finished.  I don't think it's a ton of work either, but just an idea in any case.